src/main/scala/com/typesafe/sbt/packager/distroless/DistrolessPlugin.scala

```diff
@@ -0,0 +1,262 @@
+import sbt.*
+import sbt.Keys.{discoveredMainClasses, mainClass, mappings, streams}
+
+/**
+  * ==Distroless Plugin==
+  *
+  * This plugin helps you build docker containers with distroless images
+  *
+  * ==Configuration==
+  *
+  * In order to configure this plugin take a look at the available [[com.typesafe.sbt.packager.docker.DockerKeys]]
+  *
+  * ==Requirements==
+  *
+  * You need docker to have docker installed on your system and be able to execute commands. Check with a single
+  * command:
+  *
+  * {{{
+  * docker version
+  * }}}
+  *
+  * @example
+  *   Enable the plugin in the `build.sbt`
+  *   {{{
+  *    enablePlugins(DistrolessPlugin)
+  *   }}}
+  */
+object DistrolessPlugin extends AutoPlugin {
+
+  override def requires: Plugins = DockerPlugin
+
+  object autoImport extends Keys
+
+  import DistrolessPlugin.autoImport.*
+  import DockerPlugin.autoImport.*
+
+  override def projectConfigurations: Seq[Configuration] = Seq()
+
+  // Some of the default values are now provided in the global setting based on
+  // sbt plugin best practice: https://www.scala-sbt.org/release/docs/Plugins-Best-Practices.html#Provide+default+values+in
+  override lazy val globalSettings: Seq[Setting[?]] =
+    Seq(
+      dockerBaseImage := "gcr.io/distroless/java",
+      dockerEntrypoint := Seq("/usr/bin/java"),
+      distrolessDebuggerPort := None
+    )
+
+  override lazy val projectSettings: Seq[Setting[?]] = Seq(dockerCommands := {
+    val base = dockerBaseImage.value
+    val baseDirectory = (Docker / defaultLinuxInstallLocation).value
+    val dependencies = (Docker / scriptClasspath).value
+    val debugPort = distrolessDebuggerPort.value
+    val compileMainClass = (Compile / mainClass).value
+    val compileDiscoveredMainClasses = (Compile / discoveredMainClasses).value
+    val log = streams.value.log
+
+    val layerMappings = (Docker / dockerLayerMappings).value
+    val layerIdsAscending = layerMappings.map(_.layerId).distinct.sortWith { (a, b) =>
+      // Make the None (unspecified) layer the last layer
+      a.getOrElse(Int.MaxValue) < b.getOrElse(Int.MaxValue)
+    }
+
+    val cmd = if (dockerCmd.value.isEmpty) {
+      makeClasspath(baseDirectory, dependencies) ++
+        debugPort.map(makeJdwpDebugger).toSeq ++
+        defineMainClass(compileMainClass, compileDiscoveredMainClasses, log).toSeq
+    } else {
+      dockerCmd.value
+    }
+
+    (
+      makeFromAs(base, "mainstage") +:
+        makeMaintainer((Docker / maintainer).value).toSeq :+
+        makeWorkdir(baseDirectory)
+    ) ++
+      layerIdsAscending.map { layerId =>
+        makeCopyLayerDirect(layerId, baseDirectory)
+      } ++
+      dockerLabels.value.map(makeLabel) ++
+      dockerEnvVars.value.map(makeEnvVar) ++
+      makeExposePorts(dockerExposedPorts.value, dockerExposedUdpPorts.value, distrolessDebuggerPort.value) ++
+      Seq(makeEntrypoint(dockerEntrypoint.value), makeCmd(cmd))
```

Comment:
The commands defined by the plugin are a subset of the ones available on `DockerPlugin`.
These are the most important changes:
- There is not multistage build. Since this was based on `DockerPermissionStrategy`, which in some cases it requires a RUN while building, it doesn't make sense in this context as there is no way to run anything in a distroless image to change permissions.
- There are no exposed volumes. Currently that's implemented with several RUN commands. I guess that if it would be desired, then defining just `VOLUME` without the RUNs would be possible.
- There are no `makeUser` or USER. This is due to how users are provided/created by the base distroless image. There are different tags for root and nonroot users.